src/chapters/http/http-responses.rst

```diff
@@ -0,0 +1,125 @@
+Responses
+=========
+
+.. index::
+   single: HTTP; response
+
+Each HTTP request that reaches a web server results in an HTTP response to the client.
+
+A generic HTTP response looks like this:
+
+::
+
+   HTTP/2.0 200 OK
+   Date: Wed, 22 May 2019 17:36:50 GMT
+   Content-Type: text/html; charset=UTF-8
+   Content-Length: 8050
+   Last-Modified: Wed, 22 May 2019 17:33:45 GMT
+
+   <!DOCTYPE html>
+   <html>
+   <head>
+   <meta charset="UTF-8">
+   <!--Rest of HTML page -->
+   </html>
+
+The structure has these components:
+
+.. index::
+   single: HTTP; response code
+   single: HTTP; response headers
+   single: HTTP; response line
+
+#. **Status line:** The first line of the response is the **response line**, which contains status information about the response including the **response code**. In this example, the response code is 200, which indicates the request was fulfilled successfully.
+#. **Response headers:** Below the response line are the **response headers**. Similar to request headers, these are key-value pairs that contain metadata about the response.
+#. **Blank line:** This signifies the end of the response headers.
+#. **Response body (Optional):** Below the blank line, the request body takes up the remainder of the HTTP response. This is usually HTML, CSS, other coding language files, etc.
+
+Response Codes
+--------------
+
+HTTP **response codes** are standardized codes that servers use to convey the result of attempting to fulfill the client's request. They are always three-digit numbers that fall into one of five categories based on the first digit.
+
+- 1xx (Informational): The request was received but processing has not finished
+- 2xx (Successful): The request was valid and the server successfully responded
+- 3xx (Redirection): The client should go elsewhere to access the requested resource
+- 4xx (Client Error): There was a problem with the client's request
+- 5xx (Server Error): The client's request was valid, but the server experienced an error when fulfilling it
+
+Specific codes will have all 3 digits specified, such as 201, 302, or 404. Each specific code has a specific meaning. One of the most commonly experienced error codes is 404. You have likely encountered a message like this at some point:
+
+.. figure:: figures/404.png
+   :alt: A generic 404 message displayed in a browser.
+   :height: 250px
+
+A 404 response code indicates that the requested resource does not exist on the server. This can occur when, for example, you make a typo when typing a URL into the address bar. Referring back to our postal service analogy, a 404 is similar to receiving a letter marked "Return to Sender" because the addressee doesn't live there anymore. 
+
+We don't expect you to memorize all of the response codes, but you should be able to quickly recall the most common codes.
```

Comment:
Maybe add a link to all the codes?  like in response header below?  Although there is a link below.  Would it be possible to put the link here and reference it?